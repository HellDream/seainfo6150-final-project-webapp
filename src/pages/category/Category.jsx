import React, { useState, useEffect } from 'react';
import styles from './Category.module.css';
import Header from '../../components/headers/Header.jsx';
import ProductList from '../../components/ProductList/ProductList';
import Footer from '../../components/footer/Footer';
import { isEmpty } from 'lodash';
import LoadingProgress from '../../components/progress/LoadingProgress';
import Error from '../../Error/Error';

const CategoryPage = (props) => {
    const [products, setProducts] = useState();
    const [hasError, setHasError] = useState(false);
    useEffect(() => {
        const category = props.categoryTitle;
        document.title =
            'Thrift Store - ' +
            category.charAt(0).toUpperCase() +
            category.slice(1);
        const fetchData = async () => {
            const res = await fetch(
                'https://us-central1-seainfo6150-final-project.cloudfunctions.net/api/getProducts',
                {
                    method: 'POST',
                    headers: {
                        'Content-Type': 'application/json',
                    },
                    body: JSON.stringify({ category: category }),
                }
            );
            if (res.status !== 200) {
                setHasError(true);
            } else {
                const json = await res.json();
                setProducts(json);
            }
        };

        if (isEmpty(products) && !hasError) {
            fetchData();
        }
    }, [products, hasError, props.categoryTitle]);

    if (hasError) {
        return <Error />;
    }

    return isEmpty(products) ? (
        <LoadingProgress />
    ) : (
        <>
            <Header
                back={() => {
                    props.history.goBack();
                }}
            />
            <section className={styles.section}>
                <div className={styles.container}>
                    <div className={styles.block}></div>

                    <div className={styles.headLineContainer}>
                        <h1 className={styles.headLine}>
                            {props.categoryTitle}
                        </h1>
                    </div>
                    <ProductList
                        categoryTitle={props.categoryTitle}
                        products={products}
                    />
                </div>
            </section>
            <Footer />
        </>
    );
};

export default CategoryPage;
