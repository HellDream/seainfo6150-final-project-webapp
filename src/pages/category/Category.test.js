import React from 'react';
import Category from './Category';
import { render } from '@testing-library/react'

import img from '../../images/books.jpg';
const category = {
    slug: 1,
    imageURI: img,
    title: 'books',
};
describe('CategoryItem tests', () => {
    it('renders correctly', () => {
        const { container } = render(<Category categoryTitle="books" />);
        expect(container).toMatchSnapshot();
    });
});
  