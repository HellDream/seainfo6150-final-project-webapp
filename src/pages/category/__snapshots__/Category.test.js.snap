// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryItem tests renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="loader"
    />
  </div>
</div>
`;
